src/tensor/diag.jl

```diff
@@ -32,12 +32,13 @@ Base.eltype(::Type{<:Diag{T}}) where {T} = eltype(T)
 Base.convert(::Type{<:Diag{T}},D::Diag) where {T} = Diag{T}(data(D))
 
 Base.similar(D::Diag{T}) where {T} = Diag{T}(similar(data(D)))
+Base.similar(D::Diag{T}, ::Type{T}) where {T} = Diag{T}(similar(data(D)))
```

Comment:
I am also kind of confused by this definition, it seems strange to have a similar method only defined on making the same type (i.e. the second input doesn't seem to do anything). What is this for?